src/metabase/integrations/ldap/default_implementation.clj

```diff
@@ -114,37 +113,26 @@
       flatten
       set))
 
-(defn updated-name-part
-  "Given a first or last name returned by LDAP, and the equivalent name currently stored by Metabase, return the new
-  name that should be stored by Metabase."
-  [ldap-name mb-name]
-  (if (and mb-name (nil? ldap-name))
-    ;; Don't overwrite a stored name if no name was returned by LDAP
-    mb-name
-    (or ldap-name (trs "Unknown"))))
-
 (defenterprise-schema fetch-or-create-user! :- (class User)
   "Using the `user-info` (from `find-user`) get the corresponding Metabase user, creating it if necessary."
   metabase-enterprise.enhancements.integrations.ldap
   [{:keys [first-name last-name email groups]} :- i/UserInfo
    {:keys [sync-groups?], :as settings}        :- i/LDAPSettings]
-  (let [user (db/select-one [User :id :last_login :first_name :last_name :is_active]
-                            :%lower.email (u/lower-case-en email))
+  (let [user     (db/select-one [User :id :last_login :first_name :last_name :is_active]
+                   :%lower.email (u/lower-case-en email))
         new-user (if user
                    (let [old-first-name (:first_name user)
                          old-last-name  (:last_name user)
-                         new-first-name (updated-name-part first-name old-first-name)
-                         new-last-name  (updated-name-part last-name old-last-name)
                          user-changes   (merge
-                                          (when-not (= new-first-name old-first-name) {:first_name new-first-name})
-                                          (when-not (= new-last-name old-last-name) {:last_name new-last-name}))]
+                                          (when (and first-name (not= first-name old-first-name)) {:first_name first-name})
+                                          (when (and last-name (not= last-name old-last-name)) {:last_name last-name}))]
```

Comment:
Does it still make sense to avoid setting `nil` in the DB if there's already a value?

The original reasoning was if you don't have a name set in LDAP, it'll default to `Unknown` and then you can go in and edit it yourself. And we don't want to override that.

But you can't edit your name anymore if you've logged in with SSO, right? So maybe it would make sense to just set the LDAP value in all cases?

Reply:
I thought so too, actually. There are a few tests that failed when I didn't have this change. https://github.com/metabase/metabase/blob/1aa9565c451fb2c92c4cab46c45dde7a0adab174/test/metabase/integrations/ldap_test.clj#L151-L157

Which imply a need to keep the app-db data around, although I think it's a fair question whether or not these tests are still relevant at all.

Reply:
Right, those are testing the old behavior. My question is about whether that behavior is still relevant. Probably a question for product.